arber-salihu/loGIStics
# Issue 52 and 45

This branch started as issue 52 but later on I added 45.

Issue 52: UGC Zone Scraper

Created the UgcZoneScraper.java class which is called in the processGeoJsonResponse method found in the NWSDataService.java class. When NWSDataService processes an alert it will use UgcZoneScraper to make sure all UGC Zones related to those alerts currently persist in the database. If they do not exist, UgcZoneScraper makes additional API calls gathering all missing UGC Zone geospatial data and puts the complete UGC Zone data into the database.

This required the following:

- Altering the UgcZone model
- Creating the UgcZoneService class (and tests)
- Creating the UgcZoneScraper class
- Adding minor functionality to the NWSDataService

Issue 45: Adding CRUD functionality for Geospatial Data

Researched and implemented functionality that allows for backend database operations to add geospatial data to the PostGIS database. Most of the time taken was implementing JTS Geometry classes and resolving dependency issues between JTS and Hibernate.

Test:

Currently there are no tests for the UgcZoneScraper. You can test in a more practical way by clearing the ugc_zone tables, running the program, and then looking for inserted data. Not a great option, but its what we got.

1. Clear data in the database. `TRUNCATE TABEL ugc_zone, ugc_alert, ugc_boundary;`
2. Run program
3. Check for inserted data. `SELECT ugc_code, ST_AsText(ST_Envelope(the_geom)) FROM ugc_zone;` this will show a small amount of data, but proves it is inserted.
